### ISSUE TITLE
Search Results Issue 

### ISSUE THREAD

**Issue by Taurean75** [lutostag/webextension-formsave#16]
It return results that don't contain the search term.

![search](https://user-images.githubusercontent.com/8365855/37050251-3d34fba0-2128-11e8-8867-d915f60df259.png)

**Comment by lutostag** [comment]
Yeah, thanks for making a bug on this. I am not entirely happy with the results myself either. I am using the library `fuzzy-search` https://github.com/wouter2203/fuzzy-search which matches text in order, without care for word boundaries. I was trying to come up with an approach that would work for some mis-typed words or urls that are strange. So it is "working" according to the library being used, but I agree is isn't what someone would first expect and the experience is less than ideal.

I spent the last few hours looking for a better library that can do simple stemmed, tokened, partial + fuzzy words searching, but I haven't found anything that jumps out as the perfect solution to the problem.

Two of the possibilities are:
Fuse.js
https://github.com/bvaughn/js-search

**Comment by Taurean75** [comment]
Can't you use SQL?

**Comment by lutostag** [comment]
Maybe...

I am currently using the storage api https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/local

It may be possible to use https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API but I haven't worked with it before and it does seem like a big and powerful beast :)

I'll add it to the list to look at. Or did you have something else in mind? I am not up on all the latest js tech, so maybe there is something I am missing.

**Comment by Taurean75** [comment]
I could probably help with SQL, but I think it'd be best to use an existing API.

**Comment by Taurean75** [comment]
See https://github.com/localForage/localForage

**Comment by Taurean75** [comment]
Local Storage seems to be the best for your extension since it's provided by Mozilla.

**Comment by Taurean75** [comment]
Local Storage API doesn't support SQL, right?

**Comment by Taurean75** [comment]
It does... https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Thunderbird_extensions/HowTos/Common_Thunderbird_Extension_Techniques/Use_SQLite

**Comment by lutostag** [comment]
Yeah I found that page too. Unfortunately it looks like that was only available for the old addons that are no longer supported. :/

**Comment by Taurean75** [comment]
You mean its API? Weird.

You'd look up how "cookies.sqlite" is handled.

**Comment by Taurean75** [comment]
http://kb.mozillazine.org/Cookies.sqlite

And improvise.

**Comment by Taurean75** [comment]
https://github.com/localForage/localForage looks good to me.
